src/eureka/S5_lightcurve_fitting/plots_s5.py

```diff
@@ -793,3 +797,198 @@ def plot_GP_components(lc, model, meta, fitter, isTitle=True):
         fig.savefig(meta.outputdir+fname, bbox_inches='tight', dpi=300)
         if not meta.hide_plots:
             plt.pause(0.2)
+
+
+def plot_fleck_star(lc, model, meta, fitter):
+    
+    # find number of spots
+    spotinds = [s for s in model.parameters.dict.keys() if 'spotrad' in s]   
+    counter = [s for s in spotinds if '_' in s]
+    nspots = len(spotinds) - len(counter)
+
+    # find number of limb darkening parametesr
+    limbdarks = [s for s in model.parameters.dict.keys() if 'u' in s] 
+    counter = [s for s in limbdarks if '_' in s]
+    nus = len(limbdarks) - len(counter)
+    
+    spotrad = np.zeros((lc.nchannel_fitted, nspots))
+    spotlat = np.zeros((lc.nchannel_fitted, nspots))
+    spotlon = np.zeros((lc.nchannel_fitted, nspots))
+    spotcon = np.zeros((lc.nchannel_fitted))
+    starrot = np.zeros((lc.nchannel_fitted))
+    starinc = np.zeros((lc.nchannel_fitted))
+
+    uarray = np.zeros((lc.nchannel_fitted, nus))
+
+    # Set spot/fleck parameters
+    # set a star rotation for the star object
+    # not actually used in fast mode. 
+    # overwritten if user supplies and runs slow mode
+    starrot[:] = 100 
+
+    pid_id = ''
+
+    for channel in lc.fitted_channels:
+
+        # Initialize planet
+        pl_params = PlanetParams(model, 0, channel)
+
+        # channel ID
+        if channel == 0:
+            channel_id = ''
+        else:
+            channel_id = f'_{channel}'
+
+        for n in range(nspots):
+            # read radii
+            item0 = 'spotrad' + str(n) + pid_id
+            if model.parameters.dict[item0][1] == 'free':
+                item0 += channel_id
+            value = model.parameters.dict[item0][0]
+            spotrad[channel, n] = value
+            # read latitudes
+            item0 = 'spotlat' + str(n) + pid_id
+            if model.parameters.dict[item0][1] == 'free':
+                item0 += channel_id
+            value = model.parameters.dict[item0][0]
+            spotlat[channel, n] = value
+            # read longitudes
+            item0 = 'spotlon' + str(n) + pid_id
+            if model.parameters.dict[item0][1] == 'free':
+                item0 += channel_id
+            value = model.parameters.dict[item0][0]
+            spotlon[channel, n] = value
+        # read contrasts (same for all spots)
+        item0 = 'spotcon0' + pid_id
+        if model.parameters.dict[item0][1] == 'free':
+            item0 += channel_id
+        value = model.parameters.dict[item0][0]
+        spotcon[channel] = value
+        # read stellar inclination
+        item0 = 'spotstari' + pid_id
+        if model.parameters.dict[item0][1] == 'free':
+            item0 += channel_id
+        value = model.parameters.dict[item0][0]
+        starinc[channel] = value
+        # read stellar rotation (if provided)
+        if 'spotrot' in model.parameters.dict.keys():
+            starrot[channel] = model.parameters.dict['spotrot'][0]
```

Comment:
Rather than having this code duplicated across the starry, fleck, and plotting codes, this should be consolidated into a function that gets re-used somewhere.

Reply:
It is different between fleck/starry (fleck only has one contrast, starry can have multiple), so there would be two versions of it regardless